File: nittyGritty/src/watch/my/beans/Feast.java
package watch.my.beans;

import java.util.ArrayList;

public class Feast 
{

    private ArrayList<String> meals;
    
    public Feast() 
    {
        this.meals = new ArrayList<String>();
    }

    public String addMeal(String meal)
    {
	    	boolean exists = false;
	    	
	    	for (String var1 : this.meals) 
	    		{
	            if ( var1 ==  meal)
	            	{
	            	exists = true;
	            	break;
	            	}
	            	
	        }
	    		if (exists) {return "Already available";}
	    					else
	    					{ this.meals.add(meal);
	    					return "Bigger feast";
	    					}
    	}
    
    public void whatToFeastOn()
    {
     	for (String var1 : this.meals) 
		{
     		System.out.println(var1);
        	}
    	
    }
    
    public void feastOut()
    {
    			this.meals.clear();
    }
    

}

File: nittyGritty/src/watch/my/beans/scale/Feast.scala
package watch.my.beans.scale

import scala.collection.mutable.ArrayBuffer

class Feast 
{
 
        var meals = ArrayBuffer[String]();
        
        def addMeal (meal: String)
        {
             if( !meals.contains(meal) ) meals.+=(meal) else println("Served arleady")
        }
        
        def whatToFeastOn()
        {
          println(meals)
        }

}